test/smoketest/src/smoketest/core.clj

```diff
@@ -0,0 +1,48 @@
+(ns smoketest.core
+  (:require [clojure.tools.nrepl :as nrepl]
+            [clojure.tools.nrepl.server :as server])
+  (:gen-class))
+
+;; see cider-nrepl issue #447
+(defn nrepl-handler []
+  (require 'cider.nrepl)
+  (ns-resolve 'cider.nrepl 'cider-nrepl-handler))
+
+(defn nrepl-version
+  "Launch nrepl server and send :cider-version message. This confirms
+  the repl launched with the middleware."
+  []
+  (with-open [serv (server/start-server :handler (nrepl-handler))]
```

Comment:
Perhaps starting the server shouldn't be in this function in case we decide to extend the test down the road. Also - in the beginning I didn't even see where it was being started as I didn't expect it to be in this function.

Reply:
yes ok, I'll refactor.